zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/RecordExporter.java

```diff
@@ -52,18 +54,25 @@ boolean export() {
       return true;
     }
 
+    final ValueType valueType = typedEvent.getValueType();
+    exporterMetrics.exportingLatency(
+        valueType, typedEvent.getTimestamp(), ActorClock.currentTimeMillis());
```

Comment:
:x: this must use the new `StreamClock` to prevent wrong metrics in case the engine clock was modified. This clock is already available (as an `InstantSource`) in the `ExporterDirector`, you just need to forward it so that we can use it here instead of `ActorClock`.

Reply:
I think the metrics should use System clock, because we need the exact duration. We had this issue in StreamProcessor metrics where the processing duration was always 0 because  the ActorClock is not updated with in a task.

Reply:
Interesting 🤔 Thanks for pointing out.

Reply:
Maybe I just use the timer api, instead of calculating on my onw.

Reply:
I don't think that works, you need to use the record timestamp and compare it to the "current" clock. And the "current" clock needs to be the controllable one you get as `InstantSource` in the director.

Reply:
Ah yeah, was already somewhere else with my head :) to many (benchmarks/metrics) at once.

Reply:
> I think the metrics should use System clock, because we need the exact duration

Not if we use the record timestamp as reference. Because that timestamp is not given by system time but the controllable clock

Reply:
BTW I haven't looked into the feature, but is this controllable clock still enhancing after set/moved? Or is it pinned?

Reply:
The implementation supports pinning and offset but the rest (like state, API etc.) will most likely only support pinning.

Reply:
@lenaschoenburg I was a bit confused. In my branch the clock was not even available. :D I had to rebase now an found it. 👍🏼 

I used it in the record exporter now. Be aware I still collect the metric before, so even if it is failing as otherwise I might have no metric at all. In this case I increasing latency which is what I want. I added a comment in code as well.